Sources/ENCore/App/Features/ExposureController/ExposureController.swift

```diff
@@ -142,48 +134,50 @@ final class ExposureController: ExposureControlling, Logging {
         }
     }
 
-    func updateWhenRequired() -> AnyPublisher<(), ExposureDataError> {
+    func updateWhenRequired() -> Completable {
 
         logDebug("Update when required started")
 
         if let updateStream = updateStream {
             // already updating
             logDebug("Already updating")
-            return updateStream.share().eraseToAnyPublisher()
+            return updateStream
         }
 
         let updateStream = mutableStateStream
             .exposureState
-            .first()
-            .setFailureType(to: ExposureDataError.self)
-            .flatMap { (state: ExposureState) -> AnyPublisher<(), ExposureDataError> in
+            .take(1)
+            .flatMap { (state: ExposureState) -> Completable in
                 // update when active, or when inactive due to no recent updates
                 guard [.active, .inactive(.noRecentNotificationUpdates), .inactive(.pushNotifications), .inactive(.bluetoothOff)].contains(state.activeState) else {
                     self.logDebug("Not updating as inactive")
-                    return Just(())
-                        .setFailureType(to: ExposureDataError.self)
-                        .eraseToAnyPublisher()
+                    return .empty()
                 }
 
                 self.logDebug("Going to fetch and process exposure keysets")
-                return self.fetchAndProcessExposureKeySets()
+                return .create { observer -> Disposable in
+                    self.fetchAndProcessExposureKeySets().subscribe { _ in
+                        return observer(.completed)
+                    }
+                }
             }
-            .handleEvents(
-                receiveCompletion: { _ in self.updateStream = nil },
-                receiveCancel: { self.updateStream = nil }
-            )
-            .eraseToAnyPublisher()
+            .do(onError: { [weak self] _ in
+                self?.updateStream = nil
+            }, onCompleted: { [weak self] in
+                self?.updateStream = nil
+            })
+            .asCompletable()
 
         self.updateStream = updateStream
-        return updateStream.share().eraseToAnyPublisher()
+        return updateStream
```

Comment:
Is removing the shared aspect of the stream intentional?

Reply:
We were on the fence about that too. There doesn't seem to be a share option on a Completable. Do you know of a way to get this in a way that is compatible with Completables?

Reply:
Would something like this be possible?
```
.do(onError: {}
    onCompleted: {})
.share()
.asCompletable()
```

Reply:
ah you're absolutely right. In this case your example is possible. In other cases I had a harder time bringing this back. Like in requestDiagnosisKeys() (your other comment). I'll make a ticket to review all these places where we removed share.

Reply:
👍 Sounds good! In most cases there should be a point in the chain where sharing is possible.